### PR TITLE
Settings - Add string validator for color-hex format

### LINKED ISSUE #105854: Screencast: return mouse indicator entry back to "red" after invalid entry, or add warning
Testing #105721

The `Mouse Indicator Color` setting functions properly when I enter an invalid color: it defaults back to a red indicator. But I was thinking perhaps adding a warning message to users that they have entered an invalid color (like the warning for invalid `Mouse Indicator Size`), and/or auto-filling the entry back to red would emphasize to them why their indicator has turned red again.

### PR DESCRIPTION
**Pull request by lszomoru**
This PR fixes #105854

### CODE REVIEW

File: src/vs/workbench/browser/actions/developerActions.ts [hunk 1]
```diff
@@ -324,8 +324,6 @@ configurationRegistry.registerConfiguration({
 			type: 'string',
 			format: 'color-hex',
 			default: '#FF0000',
-			minLength: 4,
-			maxLength: 9,
 			description: nls.localize('screencastMode.mouseIndicatorColor', "Controls the color in hex (#RGB, #RGBA, #RRGGBB or #RRGGBBAA) of the mouse indicator in screencast mode.")
 		},
```
Comment (chrisdias) [inline src/vs/workbench/browser/actions/developerActions.ts:328]: here is a comment

File: src/vs/workbench/services/preferences/common/preferencesValidation.ts [hunk 2]
```diff
@@ -3,9 +3,10 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
+import * as nls from 'vs/nls';
 import { JSONSchemaType } from 'vs/base/common/jsonSchema';
+import { Color } from 'vs/base/common/color';
 import { isArray } from 'vs/base/common/types';
```
Comment (chrisdias) [inline src/vs/workbench/services/preferences/common/preferencesValidation.ts:9]: anotehr comment

**Comment by chrisdias** [comment]
uber comment

**Review by roblourens — APPROVED** [review]
Thanks!

**Pull request merged**